File: 2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int max_num = *max_element(nums.begin(), nums.end());
        int answer = 0;
        int l = 0;

        for(auto num:nums){
            if(num == max_num) l++;
            else{
                answer = max(answer, l);
                l = 0;
            }
        }

        return max(answer, l);
    }
};

File: LeetCode/2503-longest-subarray-with-maximum-bitwise-and/longest-subarray-with-maximum-bitwise-and.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        answer = 0
        max_num = max(nums)

        l = 0
        for num in nums:
            if num == max_num:
                l+=1
            else:
                answer = max(l, answer)
                l = 0
        
        return max(l, answer)